File: 2015/01/p1.cpp
// Day 1: Not Quite Lisp (Part 1) https://adventofcode.com/2015/day/1
#include <bits/stdc++.h>

using namespace std;

int main(void) {
    string input = "";
    cin >> input;

    int floor = 0;

    for (int i = 0; i < input.length(); i++) {
        if (input[i] == '(') {
            floor++;
        } else {
            floor--;
        }
    }

    cout << floor << endl;

    return 0;
}

File: 2015/04/p1.py
# Day 4: The Ideal Stocking Stuffer (Part 1) https://adventofcode.com/2015/day/4
from hashlib import md5

key_prefix: str = input().strip()

suffix_value: int = 0
key_suffix: str = ""

md5_hash: str = ""

while md5_hash[:5] != "00000":
    suffix_value += 1
    key_suffix = str(suffix_value)
    secret_key: str = key_prefix + key_suffix

    md5_hash = md5(secret_key.encode("utf-8")).hexdigest()

print(key_suffix)
print(md5_hash)
